### PR TITLE
feat(plugin): add an observable plugin

### PR DESCRIPTION
**Pull request by bigopon**
# Pull Request

## 📖 Description

Add an observable plugin, primarily targeted at rxjs https://rxjs.dev/

Usage:

```ts
import Aurelia from 'aurelia';
import { ObservableConfiguration } from '@aurelia/observable'

Aurelia.register(ObservableConfiguration).app(MyApp).start();
```

A `subscribe` binding command, and a `subscribe` binding behavior are added to enable template expression scanning for `Observable`: if an expression ends with `$`, it'll be treated as such.

Example;

```html
property binding with a subscribe command
<div textcontent.subscribe="`${obs$.x} ${obs$.y} ${obs$.i}`"></div>

text content binding with subscribe binding behavior
<strong>${obs$.x & subscribe} ${obs$.y & subscribe} ${obs$.i & subscribe}</strong>

only one expression affected
<div>${`${obs$.x} ${obs$.y} ${obs$.i}` & subscribe}</div>
```

## 👩‍💻 Reviewer Notes

TODO

## 📑 Test Plan

TODO

## ⏭ Next Steps

TODO

### CODE REVIEW

File: packages/observable/package.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    "access": "public"
+  },
+  "dependencies": {
+    "@aurelia/kernel": "2.0.0-alpha.17",
+    "@aurelia/metadata": "2.0.0-alpha.17",
+    "@aurelia/platform": "2.0.0-alpha.17",
+    "@aurelia/platform-browser": "2.0.0-alpha.17",
+    "@aurelia/runtime": "2.0.0-alpha.17",
+    "@aurelia/runtime-html": "2.0.0-alpha.17",
+    "rxjs": "^6.6.7"
```
Comment (zewa666) [inline packages/observable/package.json:51]: We use the library mainly for type hinting. As such it shouldnt be a dependency but a peer dep.

**Comment by bigopon** [comment]
Some more thoughts on this: instead of an implicit opt-in for observables with a `.subscribe`/`& subscribe` and `$` suffix syntax, what if we require explicit call, says `async`?

```html
property binding with a subscribe command
<div textcontent.subscribe="`${obs$.x} ${obs$.y} ${obs$.i}`"></div>
vs
<div textcontent.bind="`${async(obs$).x} ${async(obs$).y} ${async(obs$).i}`"></div>

text content binding with subscribe binding behavior
<strong>${obs$.x & subscribe} ${obs$.y & subscribe} ${obs$.i & subscribe}</strong>
vs
<strong>${async(obs$).x} ${async(obs$).y} ${async(obs$).i}</strong>

only one expression affected
<div>${`${obs$.x} ${obs$.y} ${obs$.i}` & subscribe}</div>
vs
<div>${`${async(obs$).x} ${async(obs$).y} ${async(obs$).i}`}</div>
```

The benefits of this is we don't need to introduce new syntax, and `async` call can be reused for dealing with promise, like the following example:

```html
<template>
  <!-- Simple usage: -->
  <h1>message: ${async(message).value}</h1>
  
  <!-- With a placeholder: -->
  <h1>message: ${async(message).value ? async(message).value : '...'}</h1>
</template>
```

cc @zewa666 @hiaux0

**Comment by hiaux0** [comment]
A simple method call instead of all the custom syntax seems actually desirable.
+1

**Comment by bigopon** [comment]
Will close this for now, if we ever do it, it'll be in a form of a method.

**Pull request closed without merging**